packages/preview-middleware-client/src/adp/dialog-factory.ts

```diff
@@ -0,0 +1,134 @@
+import Dialog from 'sap/m/Dialog';
+import UI5Element from 'sap/ui/core/Element';
+import Fragment from 'sap/ui/core/Fragment';
+import RuntimeAuthoring from 'sap/ui/rta/RuntimeAuthoring';
+
+import { getTextBundle } from '../i18n';
+
+import AddFragment, { AddFragmentOptions } from './controllers/AddFragment.controller';
+import AddTableColumnFragments from './controllers/AddTableColumnFragments.controller';
+import ControllerExtension from './controllers/ControllerExtension.controller';
+import ExtensionPoint from './controllers/ExtensionPoint.controller';
+
+import { ExtensionPointData } from './extension-point';
+
+export const enum DialogNames {
+    ADD_FRAGMENT = 'AddFragment',
+    ADD_TABLE_COLUMN_FRAGMENTS = 'AddTableColumnFragments',
+    CONTROLLER_EXTENSION = 'ControllerExtension',
+    ADD_FRAGMENT_AT_EXTENSION_POINT = 'ExtensionPoint'
+}
+
+type Controller = AddFragment | AddTableColumnFragments | ControllerExtension | ExtensionPoint;
+
+export const OPEN_DIALOG_STATUS_CHANGED = 'OPEN_DIALOG_STATUS_CHANGED';
+
+export class DialogFactory {
+    private static readonly eventTarget = new EventTarget();
+    private static isDialogOpen = false;
+    /**
```

Comment:
Shouldn't the jsdoc be above the property?

Reply:
It is for the public property `canOpenDialog`, so it is correct.